File: LivePriceBackend/DTOs/User/UserUpdateModel.cs
using LivePriceBackend.Constants.Enums;

namespace LivePriceBackend.DTOs.User;

public class UserUpdateModel
{
    public string? Name { get; set; }
    public string? Surname { get; set; }
    public string? Username { get; set; }
    public string? Email { get; set; }
    public string? Password { get; set; }
    public Role? Role { get; set; }
    public int? CustomerId { get; set; }
} 

File: live_price_frontend/lib/modules/users/models/user_update_model.dart
import 'user_role.dart';

class UserUpdateModel {
  final String? username;
  final String? password;
  final String? email;
  final String? name;
  final String? surname;
  final UserRole? role;
  final int? customerId;

  UserUpdateModel({
    this.username,
    this.password,
    this.email,
    this.name,
    this.surname,
    this.role,
    this.customerId,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
      'email': email,
      'name': name,
      'surname': surname,
      'role': role?.id,
      'customerId': customerId,
    };
  }
}
